expression/expression.go

```diff
@@ -1388,6 +1388,11 @@ func PropagateType(evalType types.EvalType, args ...Expression) {
 				newCol.(*CorrelatedColumn).RetType = col.RetType.Clone()
 				args[0] = newCol
 			}
+			if args[0].GetType().Tp == mysql.TypeNewDecimal {
+				if newDecimal > mysql.MaxDecimalScale {
+					newDecimal = mysql.MaxDecimalScale
+				}
+			}
```

Comment:
Do we need some testcase to cover it?

Reply:
https://github.com/pingcap/tics/pull/3393